src/environments/array.js

```diff
@@ -49,10 +49,8 @@ function getHLines(parser: Parser): boolean[] {
 const validateAmsEnvironmentContext = context => {
     const settings = context.parser.settings;
     if (!settings.displayMode) {
-        throw new ParseError(`{${context.envName}} cannot be used inline.`);
-    } else if (settings.strict && !settings.topEnv) {
-        settings.reportNonstrict("textEnv",
-            `{${context.envName}} called from math mode.`);
+        throw new ParseError(`{${context.envName}} can be used only in` +
+            ` display mode.`);
```

Comment:
In addition to checking `!settings.displayMode` here maybe would should also check `settings.useStrictBehavior()` as well like so:
```
if (!settings.displayMode) {
   const message = `{${context.envName}} can be used only in display mode.`;
   if (settings.useStrictBehavior("invalidEnvironment", message)) {
      throw new ParserError(message);
   }
}
```

Reply:
I dunno if this is a good idea. Most of the AMS environments have automatic numbering. That isn't likely to combine well with inline math.  I'm usually the person here who advocates permissive behavior, but I think this is a case where a `ParseError` is better than math rendered in a strange way.

Reply:
I'm going to merge this PR as is. We can still edit `validateAmsEnvironmentContext()` in a separate PR if we decide such an edit is necessary.